File: component_core/src/main/java/com/ashlikun/libcore/utils/CacheUtils.java
package com.ashlikun.libcore.utils;

import android.os.Environment;

import java.io.File;

/**
 * 作者　　: 李坤
 * 创建时间: 2018/7/18　10:50
 * 邮箱　　：496546144@qq.com
 * <p>
 * 功能介绍：
 */
public class CacheUtils {
    //app缓存路径，内部
    public static String appCachePath;
    //app文件路径，内部
    public static String appFilePath;
    //appsd卡缓存路径
    public static String appSDCachePath;
    //appsd卡文件路径
    public static String appSDFilePath;

    /**
     * @param rootName 跟目录名称，一般为app名称
     */
    public static void init(String rootName) {
        String cacheStr = "/" + rootName + "/cache";
        String fileStr = "/" + rootName + "/file";
        appCachePath = AppUtils.getApp().getCacheDir().getPath() + cacheStr;
        appFilePath = AppUtils.getApp().getFilesDir().getPath() + fileStr;
        appSDCachePath = Environment.getExternalStorageDirectory().getPath() + cacheStr;
        appSDFilePath = Environment.getExternalStorageDirectory().getPath() + fileStr;

        File file = new File(appSDCachePath);
        if (file.exists() || file.mkdirs()) {
        }
        File file2 = new File(appSDFilePath);
        if (file2.exists() || file2.mkdirs()) {
        }
    }
}

File: component_core/src/main/java/com/ashlikun/baseproject/libcore/utils/CacheUtils.kt
package com.ashlikun.baseproject.libcore.utils

import android.os.Environment

import com.ashlikun.utils.AppUtils

import java.io.File

/**
 * 作者　　: 李坤
 * 创建时间: 2018/7/18　10:50
 * 邮箱　　：496546144@qq.com
 *
 *
 * 功能介绍：
 */
object CacheUtils {
    //app缓存路径，内部
    var appCachePath: String = ""
    //app文件路径，内部
    var appFilePath: String = ""
    //appsd卡缓存路径
    var appSDCachePath: String = ""
    //appsd卡文件路径
    var appSDFilePath: String = ""

    /**
     * @param rootName 跟目录名称，一般为app名称
     */
    fun init(rootName: String) {
        val cacheStr = "/$rootName/cache"
        val fileStr = "/$rootName/file"
        appCachePath = AppUtils.getApp().cacheDir.path + cacheStr
        appFilePath = AppUtils.getApp().filesDir.path + fileStr
        appSDCachePath = Environment.getExternalStorageDirectory().path + cacheStr
        appSDFilePath = Environment.getExternalStorageDirectory().path + fileStr

        val file = File(appSDCachePath)
        if (file.exists() || file.mkdirs()) {
        }
        val file2 = File(appSDFilePath)
        if (file2.exists() || file2.mkdirs()) {
        }
    }
}
